docs/sql-migration-guide.md

```diff
@@ -215,6 +215,8 @@ license: |
   For example `SELECT timestamp 'tomorrow';`.
 
   - Since Spark 3.0, the `size` function returns `NULL` for the `NULL` input. In Spark version 2.4 and earlier, this function gives `-1` for the same input. To restore the behavior before Spark 3.0, you can set `spark.sql.legacy.sizeOfNull` to `true`.
+  
+  - Since Spark 3.0, when `array` function is called without parameters, it returns an empty array with `NullType` data type. In Spark version 2.4 and earlier, it returns an empty array with string type.
```

Comment:
Before merging this, let's make it sure that we don't need to add a legacy conf for this.
cc @gatorsmile

Reply:
Ur, keeping the legacy behaviour in the conf looks somewhat reasonable to me for hive users.